com.unity.multiplayer.mlapi/Runtime/Core/NetworkUpdateLoop.cs

```diff
@@ -0,0 +1,525 @@
+﻿using System.Linq;
```

Comment:
I don't understand why we need this file at all... seems overly complicated for just adding something to the loop, like are we just wanting any sort of object be able to register itself in the loop ?

Reply:
[From RFC's motivation section:](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/blob/master/text/0008-network-update-loop.md#motivation)

> Even though it is possible to use low-level Player Loop API directly to insert custom `PlayerLoopSystem`s into the existing player loop, it also requires a non-trivial amount of boilerplate code and could cause fundamental issues in the engine runtime if not done very carefully. **We would like to have less boilerplate code and also be less error-prone at runtime.**

> Beyond that, the proposed design standardizes `NetworkUpdateStage`s which are going to be executed at specific points in the player loop. This allows other framework systems tied into `NetworkUpdateLoop` to be aligned, such as RPCs executing at specific stages. **We would like to standardize the execution of network stages as an infrastructure in the framework.**